src/main/java/it/gov/pagopa/pu/pagopapayments/service/paymentsreporting/PaymentsReportingService.java

```diff
@@ -15,14 +17,24 @@ public class PaymentsReportingService {
 
   private final NodeForPaClient nodeForPaClient;
   private final BrokerService brokerService;
+  private final FileShareClient fileShareClient;
 
-  public PaymentsReportingService(NodeForPaClient nodeForPaClient, BrokerService brokerService) {
+  public PaymentsReportingService(NodeForPaClient nodeForPaClient, BrokerService brokerService, FileShareClient fileShareClient) {
     this.nodeForPaClient = nodeForPaClient;
     this.brokerService = brokerService;
+      this.fileShareClient = fileShareClient;
   }
 
   public List<PaymentsReportingIdDTO> getPaymentsReportingList(Long organizationId, String accessToken) {
     BrokerForNodoPaDTO brokerForNodoPaDTO = brokerService.getBrokerForNodoPaDTOByOrganizationId(organizationId, accessToken);
     return nodeForPaClient.getPaymentsReportingList(brokerForNodoPaDTO);
   }
+
+  public String uploadPaymentsReporting(Long organizationId, String reportingId, String fileName, String accessToken) {
```

Comment:
why are we asking the fileName as parameter? we should build it here and send to the client (let's add it to the client with a TODO referring to ticket P4ADEV-2107